src/Umbraco.Configuration/Models/RequestHandlerSettings.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Extensions.Configuration;
+using Umbraco.Core;
+using Umbraco.Core.Configuration.UmbracoSettings;
+
+namespace Umbraco.Configuration.Models
+{
+    internal class RequestHandlerSettings : IRequestHandlerSettings
+    {
+        private readonly IConfiguration _configuration;
+        public RequestHandlerSettings(IConfiguration configuration)
+        {
+            _configuration = configuration;
+        }
+
+        public bool AddTrailingSlash   => _configuration.GetValue<bool?>("Umbraco:CMS:RequestHandler:AddTrailingSlash") ?? true;
+        public bool ConvertUrlsToAscii   => _configuration.GetValue<string>("Umbraco:CMS:RequestHandler:ConvertUrlsToAscii").InvariantEquals("true");
+        public bool TryConvertUrlsToAscii   => _configuration.GetValue<string>("Umbraco:CMS:RequestHandler:ConvertUrlsToAscii").InvariantEquals("try");
+
+
+        //We need to special handle ":", as this character is special in keys
```

Comment:
(minor) Just to clarify, are we removing the : from the character collection as it is used as an identifier? If so, it might be good to mention in the comments e.g. "Remove instances of the ':' character, because...".

Reply:
No the issue is, if/when we use ":" as key in the json-config file. it is not understood by the framework.

There are two alternatives
1. Special handle `":"` and always replace with `""`
1. Implement the config more verbose, such that we don't use `":"` as key.
E.G
```
{
    "Umbraco": {
        "CMS": {
            "RequestHandler": {
                "AddTrailingSlash": true,
                "CharCollection": [
                    {"Char": ":", "Replacement": ""},
                    {"Char": "\"", "Replacement": ""},
                    {"Char": "'", "Replacement": ""},
                    {"Char": "%", "Replacement": ""},
                   ...
                ]
            }
        }
    }
}
```
instead of the current
```
{
    "Umbraco": {
        "CMS": {
            "RequestHandler": {
                "AddTrailingSlash": true,
                "CharCollection": {
                    ":": "",
                    "\"":"",
                    "'":"",
                    "%":"",
                   ...
                }
            }
        }
    }
}
```

Reply:
I would prefer that we don't need to hack this with special chars and rules. Although the Char/Replacement is more verbose it doesn't need any hacks so I vote for that.